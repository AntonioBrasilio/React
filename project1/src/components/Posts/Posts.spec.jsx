import { render, screen } from "@testing-library/react";
import { Posts } from ".";

const props = {
    posts: [
        {
            id: 1,
            title: "mock-title-1",
            body: "mock-body-1",
            cover: "img/mock-img-1.png",
        },
        {
            id: 2,
            title: "mock-title-2",
            body: "mock-body-2",
            cover: "img/mock-img-2.png",
        },
        {
            id: 3,
            title: "mock-title-3",
            body: "mock-body-3",
            cover: "img/mock-img-3.png",
        },
    ],
};

describe("<Posts />", () => {
    it("should render the posts properties correctly", () => {
        render(<Posts {...props} />);

        expect(
            screen.getAllByRole("heading", { name: /mock-title/i })
        ).toHaveLength(3);
        expect(screen.getAllByRole("img", { name: /mock-title/i })).toHaveLength(
            3
        );
        expect(screen.getAllByText(/mock-body/i)).toHaveLength(3);
    });

    it("should not render posts", () => {
        render(<Posts  />);
        expect(
            screen.queryByRole("heading", { name: /mock-title/i })
        ).not.toBeInTheDocument();
    });

    it("should match snapshot", () => {
        render(<Posts {...props} />);
        expect(screen.getByTestId("posts-test-id")).toMatchSnapshot();
    });
});
