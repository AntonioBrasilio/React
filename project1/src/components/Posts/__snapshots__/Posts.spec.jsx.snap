// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
  data-testid="posts-test-id"
>
  <div
    class="post"
    data-testid="post-card-test-id"
  >
    <img
      alt="mock-title-1"
      src="img/mock-img-1.png"
    />
    <div
      class="post-content"
      id="1"
    >
      <h2>
        mock-title-1
      </h2>
      <p>
        mock-body-1
      </p>
    </div>
  </div>
  <div
    class="post"
    data-testid="post-card-test-id"
  >
    <img
      alt="mock-title-2"
      src="img/mock-img-2.png"
    />
    <div
      class="post-content"
      id="2"
    >
      <h2>
        mock-title-2
      </h2>
      <p>
        mock-body-2
      </p>
    </div>
  </div>
  <div
    class="post"
    data-testid="post-card-test-id"
  >
    <img
      alt="mock-title-3"
      src="img/mock-img-3.png"
    />
    <div
      class="post-content"
      id="3"
    >
      <h2>
        mock-title-3
      </h2>
      <p>
        mock-body-3
      </p>
    </div>
  </div>
</div>
`;
